junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/ExtensionRegistry.java

```diff
@@ -57,6 +58,10 @@
 	public static ExtensionRegistry createRegistryWithDefaultExtensions() {
 		ExtensionRegistry extensionRegistry = new ExtensionRegistry(null);
 		DEFAULT_EXTENSIONS.forEach(extensionRegistry::registerDefaultExtension);
+		if (Boolean.getBoolean("org.junit.jupiter.engine.extension.ExtensionRegistry.withServiceLoader")) {
```

Comment:
This needs to be supported analogous to our support for deactivating conditions.

As such, the name of the property should not be tightly coupled to the name of the class that consumes it, and it should use `ConfigurationParameters`.

Consult the following for examples:

- `org.junit.jupiter.engine.Constants.DEACTIVATE_CONDITIONS_PATTERN_PROPERTY_NAME`.
- `org.junit.jupiter.engine.execution.ConditionEvaluator.getDeactivatePatternString(ConfigurationParameters)`

I'd recommend a property name like `junit.extensions.auto-detect`.